lib/cloud_controller/backends/runners.rb

```diff
@@ -73,31 +73,93 @@ def dea_apps(batch_size, last_id)
       query.all
     end
 
-    def dea_apps_hm9k(batch_size, last_id)
-      query = App.select_all(App.table_name).
-              runnable.
-              dea.
-              where("#{App.table_name}.id > ?", last_id).
-              order("#{App.table_name}__id".to_sym).
-              limit(batch_size).
-              eager(:latest_droplet, :latest_package, current_droplet: :package)
-
-      apps = query.all.reject { |a| a.package_state == 'FAILED' }
-
-      app_hashes = apps.map do |app|
-        {
-          'id'            => app.guid,
-          'instances'     => app.instances,
-          'state'         => app.state,
-          'memory'        => app.memory,
-          'version'       => app.version,
-          'updated_at'    => app.updated_at || app.created_at,
-          'package_state' => app.package_state,
-        }
+    def dea_apps_hm9k
```

Comment:
This method is pretty big and does a lot. It might be clearer to at least break out the queries into their own methods. You could then unit test the queries, so you don't need the comments describing them.